bothan-lib/src/worker/rest.rs

```diff
@@ -45,31 +47,62 @@ pub async fn start_polling<S: Store, E: Display, P: AssetInfoProvider<Error = E>
 
         select! {
             _ = cancellation_token.cancelled() => break,
-            r = timeout(interval.period(), provider.get_asset_info(&ids)) => handle_polling_result(r, &store).await,
+            r = poll_with_timeout(&provider, &ids, update_interval, &metrics) => handle_polling_result(r, &store, &metrics).await,
         }
     }
 }
 
+async fn poll_with_timeout<P, E>(
+    provider: &P,
+    ids: &[String],
+    timeout_interval: Duration,
+    metrics: &Metrics,
+) -> Result<Result<Vec<AssetInfo>, E>, Elapsed>
+where
+    E: Display,
+    P: AssetInfoProvider<Error = E>,
+{
+    let start_time = chrono::Utc::now().timestamp_millis();
+    let result = timeout(timeout_interval, provider.get_asset_info(ids)).await;
+    let elapsed_time = chrono::Utc::now().timestamp_millis() - start_time;
+
+    let polling_result = match result {
```

Comment:
while this works, i'd prefer if we just use a ref here instead in case we use the values inside later